charmcraft.yaml

```diff
@@ -0,0 +1,467 @@
+
+    discover-job-kube-node-selectors:
+      description: |
+        Defines one or more pod node selectors for the discover job. 
+        
+        Each k=v pair is separated by a ','. For example: key1=value1,key2=value2.
+      type: string
+
+    spec-job-kube-annotations:
+      description: |
+        Defines one or more pod annotations for the spec job. 
+        
+        Each k=v pair is separated by a ','. For example: key1=value1,key2=value2
+      type: string
+
+    check-job-kube-annotations:
+      description: |
+        Defines one or more pod annotations for the check job. 
+        
+        Each k=v pair is separated by a ','. For example: key1=value1,key2=value2
+      type: string
+
+    discover-job-kube-annotations:
+      description: |
+        Defines one or more pod annotations for the discover job. 
+        
+        Each k=v pair is separated by a ','. For example: key1=value1,key2=value2
+      type: string
+
+    ##### Logging config #####
+    storage-type:
+      description: |
+        Storage type for logs.
+
+        Acceptable values are: "MINIO", "S3" (AWS)
+      default: "MINIO"
+      type: string
+
+    storage-bucket-logs:
+      description: Name of logs storage bucket.
+      default: "airbyte-dev-logs"
+      type: string
+
+    logs-ttl:
+      description: |
+        Number of days until logs are purged from object storage.
+      default: 30
+      type: int
+
+    storage-bucket-state:
+      description: Name of state storage bucket.
+      default: "airbyte-state-storage"
+      type: string
+
+    storage-bucket-activity-payload:
+      description: Name of activity payload storage bucket.
+      default: "airbyte-payload-storage"
+      type: string
+
+    storage-bucket-workload-output:
+      description: Name of workload output storage bucket.
+      default: "airbyte-state-storage"
+      type: string
+
+    ##### Miscellaneous config #####
+    pod-running-ttl-minutes:
+      description: Number of minutes until a running job pod is removed.
+      default: 240
+      type: int
+
+    pod-successful-ttl-minutes:
+      description: Number of minutes until a successful job pod is removed.
+      default: 30
+      type: int
+
+    pod-unsuccessful-ttl-minutes:
+      description: Number of minutes until an unsuccessful job pod is removed.
+      default: 1440
+      type: int
+
+# The containers and resources metadata apply to Kubernetes charms only.
+# See https://juju.is/docs/sdk/metadata-reference for a checklist and guidance.
+
+# Your workload’s containers.
+containers:
+  airbyte-api-server:
+    resource: airbyte-api-server
+  airbyte-bootloader:
+    resource: airbyte-bootloader
+  airbyte-connector-builder-server:
+    resource: airbyte-connector-builder-server
+  airbyte-cron:
+    resource: airbyte-cron
+  airbyte-pod-sweeper:
+    resource: airbyte-pod-sweeper
+  airbyte-server:
+    resource: airbyte-server
+  airbyte-workers:
+    resource: airbyte-workers
```

Comment:
I did not find much guidance on the approach of using services vs. containers for a multi-container application. It might be possible to do this using services, but I am not sure how that would affect the charm operationally (e.g. observing logs).

We have not yet designed the scaling aspect of the Airbyte charms, but yes, the current assumption is that each juju unit/pod would contain these containers.

Reply:
Wrt. scaling, I figure it's fair game to take inspiration from the existing Helm chart.